### ISSUE TITLE
Building a dotnet new iOS app on the command line on a Mac fails with a Windows path in the nuget.config file

### ISSUE THREAD

**Issue by rolfbjarne** [dotnet/runtime#104402]
_From @brunck on Mon, 01 Jul 2024 22:46:14 GMT_

It's helpful to have a `nuget.config` file on the Windows side to shorten paths, due to the dastardly max path problem in VS for Windows. This breaks building the project on a Mac at the command line.

Building the same project in VS for Windows (paired to a Mac) works.

### Steps to Reproduce

1. `dotnet new ios -n someName` on a Mac
2. Create a `nuget.config` file for the created project with a Windows path like `C:\n`
for example:
```xml
<configuration>
  <config>
    <add key="globalPackagesFolder" value="C:\n" />
  </config>
</configuration>
```
3. `dotnet build`
4. or, just download and run the provided example project

### Expected Behavior
The app builds with no errors.

### Actual Behavior
A `Failed to create CoreCLR, HRESULT: 0x80070057` error, along with a message about setting the `PublishTrimmed` property to false, which is disallowed in a.NET for iOS app anyway

### Environment

<details>
<summary>Version information</summary>

.NET SDK  version 8.0.302
Workload info: `ios                        17.2.8053/8.0.100      SDK 8.0.300`

</details>

### Example Project (If Possible)

https://github.com/brunck/reproductions/tree/master/NuGetConfigBuildError

_Copied from original issue xamarin/xamarin-macios#20800_

**Comment by rolfbjarne** [comment]
_From @rolfbjarne on Wed, 03 Jul 2024 19:37:43 GMT_

This looks like a bug in .NET:

```shell
$ mkdir rootdir
$ cd rootdir
$ dotnet new console
[...]
$ dotnet publish /p:PublishDir=C:
[...]
$ dotnet exec C:/rootdir.dll
Failed to create CoreCLR, HRESULT: 0x80070057
```

**Comment by rolfbjarne** [comment]
_From @rolfbjarne on Wed, 03 Jul 2024 19:40:38 GMT_

Looks like variations of https://github.com/dotnet/runtime/issues/3163 and https://github.com/dotnet/runtime/issues/75387

**Comment by vitek-karas** [comment]
This is weird to me: 
```$ dotnet exec C:/rootdir.dll
Failed to create CoreCLR, HRESULT: 0x80070057
```

That is trying to run the app - I would not expect an ios app to run on CoreCLR, we could produce a better error, but functionally this is simply not supported.

But the issue states that it fails during build, which would not be this failure. What does the build failure look like?

**Comment by brunck** [comment]
```
  Determining projects to restore...
  All projects are up-to-date for restore.
  Detected signing identity:
          
    Bundle Id: com.companyname.TestiOSApp
    App Id: com.companyname.TestiOSApp
  TestiOSApp -> /Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/bin/Debug/net8.0-ios/iossimulator-x64/TestiOSApp.dll
  Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
  Optimizing assemblies for size. This process might take a while.
  Failed to create CoreCLR, HRESULT: 0x80070057
/Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/C:/n/microsoft.net.illink.tasks/8.0.6/build/Microsoft.NET.ILLink.targets(134,5)
: error MSB6006: "dotnet" exited with code 137. [/Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/TestiOSApp.csproj]            /Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/C:/n/microsoft.net.illink.tasks/8.0.6/build/Microsoft.NET.ILLink.targets(87,5):
 error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false. [/Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/TestiOSApp.csproj]                                                                          
Build FAILED.

/Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/C:/n/microsoft.net.illink.tasks/8.0.6/build/Microsoft.NET.ILLink.targets(134,5)
: error MSB6006: "dotnet" exited with code 137. [/Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/TestiOSApp.csproj]            /Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/C:/n/microsoft.net.illink.tasks/8.0.6/build/Microsoft.NET.ILLink.targets(87,5):
 error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false. [/Users/x/Documents/SourceCode/reproductions/NuGetConfigBuildError/TestiOSApp.csproj]                                                                              0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:04.20

Workload updates are available. Run `dotnet workload list` for more information.
```

**Comment by vitek-karas** [comment]
It fails to launch illink.

**Comment by rolfbjarne** [comment]
Here's a simpler repro: https://github.com/dotnet/runtime/issues/104402#issuecomment-2207072847

**Comment by vitek-karas** [comment]
The command which fails looks something like
```
dotnet ~/repro/app/C:/n/nugetcatchpath/illink.dll
```

The problem is that this leads to `~/repro/app/C:/n/nugetcatchpath/illink.dll` being part of TPA. But TPA is a string which stores a list of file paths which are delimited by the OS's path separator, which is `:` on mac. So the path is seen as two by the runtime:
```
~/repro/app/C
/n/nugetcatchpath/illink.dll
```

Populating the internal TPA cache with this will fail because neither points to an existing file.

The short version of this bug is that running a .NET app in a path which has ':' in it somewhere fails.

**Comment by elinor-fung** [comment]
Split the `dotnet publish` scenario into a separate issue in SDK: https://github.com/dotnet/sdk/issues/49671
